lib/jxl/modular/encoding/encoding.h

```diff
@@ -27,7 +27,7 @@ struct ANSCode;
 class BitReader;
 
 // Valid range of properties for using lookup tables instead of trees.
-constexpr int32_t kPropRangeFast = 512;
+constexpr int32_t kPropRangeFast = 512 << 6;
```

Comment:
This causes the table to occupy at least 32kB of stack size, which is probably too much.

Reply:
Agreed, it's too much. The table size currently is `6*kPropRangeFast` bytes, so that's even 196kB. It still worked reasonably well on my machine (I benchmarked reasonable speeds) but I assume it is not a good idea to have a table that large.

I'll cap the shifting to << 4, and also specialize the lookup table for the case of zero offsets and also adjust the encoder size specialization.

The table has then size `2*kPropRangeFast` instead of  `6*kPropRangeFast` for both encode and decode — in encode we had uint16 context_lookup and uint8 offsets (no multipliers), in decode we had uint8 context_lookup and uint8 offsets and multipliers.

Still it will be 16kB then instead of 3kB because `kPropRangeFast` is 16x as large as before. I'll benchmark this and put some results in the main PR discussion.

Reply:
Sorry, I somehow missed this.

The stack size issue is not a question of speed but of portability - I think 16kB should be fine though.

Reply:
We could also allocate a buffer per-group instead... Unclear if that'd be a significant improvement.

Reply:
Another approach could be to detect that the cutoffs were shifted (they all have N lsb zero bits) and shift the property values before using the lookup table. Adds a variable shift but reduces memory — possibly we can even get away with a smaller table in the 8-bit case too.

But for now let's just keep it simple and do it like this, unless there's an issue with it on some platform or something.